xds/googledirectpath/googlec2p_test.go

```diff
@@ -251,3 +252,15 @@ func TestBuildXDS(t *testing.T) {
 		})
 	}
 }
+
+// TestBuildFailsWhenCalledWithAuthority asserts that c2p resolver.build
+// fails when called with a URL with an authority.
```

Comment:
This will probably have to change a little now. Maybe:

```golang
// TestDialFailsWhenTargetContainsAuthority attempts to Dial a target URI of
// google-c2p scheme with a non-empty authority and verifies that it fails with 
// an expected error.
```

Reply:
ive updated this as well